### ISSUE TITLE
    raise Exception('Invalid merge mode: ' + str(mode)) Exception: Invalid merge mode: abs_sub

### ISSUE THREAD

**Issue by amitamisra** [fchollet/keras#4448]
I am trying to do sentence similarity using CNN+LSTM.

I get an invalid merge when merging the 2 models.
    
   model_sen1 = Sequential()
    embedding_layer= create_embedding_matrix(word_index)
    model_sen1.add(embedding_layer)
    model_sen1.add(LSTM(64, return_sequences=True))
    model_sen1.add(Dropout(0.2))
    model_sen1.add(Convolution1D(64 // 2, 5, border_mode="valid"))
    model_sen1.add(MaxPooling1D(pool_length=2, border_mode="valid"))
    model_sen1.add(Dropout(0.2))
    
    
    model_sen2 = Sequential()
    model_sen2.add(embedding_layer)
    model_sen2.add(LSTM(64, return_sequences=True))
    model_sen2.add(Dropout(0.2))
    model_sen2.add(Convolution1D(64 // 2, 5, border_mode="valid"))
    model_sen2.add(MaxPooling1D(pool_length=2, border_mode="valid"))
    model_sen2.add(Dropout(0.2))
    
    l_mul = Sequential()
    l_mul.add(Merge([ model_sen1,  model_sen2], mode='mul'))

    l_sub = Sequential()
    l_sub.add(Merge([ model_sen1,  model_sen2], mode='abs_sub'))

    model = Sequential()
    model.add(Merge([l_mul, l_sub], mode='concat', concat_axis=-1))
    model.add(Dropout(0.2))
    model.add(Dense(1, init='normal'))
    
    model.add(Activation('relu')) 
    model.summary()
    model.compile(loss='mean_squared_error', optimizer='rmsprop')
    return model

**Comment by farizrahman4u** [comment]
Its obvious from the message. There is no mode called abs_sub.

**Comment by amitamisra** [comment]
It was supported earlier. I am new to keras. Is there a way to do it using lambda function?

**Comment by farizrahman4u** [comment]
```python
l_sub.add(Merge([ model_sen1,  model_sen2], mode=lambda x: K.abs(x[0] - x[1]), output_shape=lambda s: s[0]))
```

**Comment by amitamisra** [comment]
It worked. Thanks.
